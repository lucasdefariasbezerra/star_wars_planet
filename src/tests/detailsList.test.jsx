import React from 'react';
import DetailsList from '../components/shared/detailsList';
import { shallow } from 'enzyme';

let wrapper;
const props = {
    list: [ { title: 'test', director: 'lewis test', producer: 'lewis producer' },
            { title: 'test', director: 'lewis test', producer: 'lewis producer' }]
};

describe('</DetailsList>', () => {
    beforeEach(() => {
        wrapper = shallow(
            <DetailsList { ...props} />
        );
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have length of two', () => {
        expect(wrapper.find('ListItem')).toHaveLength(2);
    });
});