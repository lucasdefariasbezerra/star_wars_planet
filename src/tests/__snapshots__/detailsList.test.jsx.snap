// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</DetailsList> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DetailsList
    list={
      Array [
        Object {
          "director": "lewis test",
          "producer": "lewis producer",
          "title": "test",
        },
        Object {
          "director": "lewis test",
          "producer": "lewis producer",
          "title": "test",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ListItem
          item={
            Object {
              "director": "lewis test",
              "producer": "lewis producer",
              "title": "test",
            }
          }
        />,
        <ListItem
          item={
            Object {
              "director": "lewis test",
              "producer": "lewis producer",
              "title": "test",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "test",
        "nodeType": "function",
        "props": Object {
          "item": Object {
            "director": "lewis test",
            "producer": "lewis producer",
            "title": "test",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "test",
        "nodeType": "function",
        "props": Object {
          "item": Object {
            "director": "lewis test",
            "producer": "lewis producer",
            "title": "test",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ListItem
            item={
              Object {
                "director": "lewis test",
                "producer": "lewis producer",
                "title": "test",
              }
            }
          />,
          <ListItem
            item={
              Object {
                "director": "lewis test",
                "producer": "lewis producer",
                "title": "test",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "test",
          "nodeType": "function",
          "props": Object {
            "item": Object {
              "director": "lewis test",
              "producer": "lewis producer",
              "title": "test",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "test",
          "nodeType": "function",
          "props": Object {
            "item": Object {
              "director": "lewis test",
              "producer": "lewis producer",
              "title": "test",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
